### PR TITLE
feature/#1805 added checking if entity exists

### LINKED ISSUE #1805: Protect the Reason APIs in case the entity or its ID does no exist
If the entity does not exist, the ReasonCollection should be False, and a specific Reason should be provided. 

Must be propagated to enterprise and documentation.

### PR DESCRIPTION
**Pull request by toan-quach**
#1805

### CODE REVIEW

File: taipy/core/job/_job_manager.py [hunk 1]
```diff
@@ -92,9 +92,14 @@ def _is_deletable(cls, job: Union[Job, JobId]) -> ReasonCollection:
         reason_collector = ReasonCollection()
 
         if isinstance(job, str):
-            job = cls._get(job)
+            job_id = job
+            job = cls._get(job, None)
```
Comment (jrobinAV) [inline taipy/core/job/_job_manager.py:96]:
```suggestion
            if job:= cls._get(job, None) is None:
                reason_collector._add_reason(job_id, EntityDoesNotExist(job_id))
                return reason_collector
```

File: taipy/core/job/_job_manager.py [hunk 2]
```diff
@@ -92,9 +92,14 @@ def _is_deletable(cls, job: Union[Job, JobId]) -> ReasonCollection:
...
+        else:
+            job_id = job.id
```
Comment (jrobinAV) [inline taipy/core/job/_job_manager.py]: I believe we don't need that.
```suggestion
```

File: taipy/core/job/_job_manager.py [hunk 3]
```diff
@@ -92,9 +92,14 @@ def _is_deletable(cls, job: Union[Job, JobId]) -> ReasonCollection:
...
 
-        if job and not job.is_finished():
+        if job is None:
+            reason_collector._add_reason(job_id, EntityDoesNotExist(job_id))
+        elif not job.is_finished():
```
Comment (jrobinAV) [inline taipy/core/job/_job_manager.py]: From the previous comment, the case where the job can be None is already handled.
```suggestion
        if not job.is_finished():
```

**Review by jrobinAV — DISMISSED** [review]
The logic is there but I just proposed a change in the way we handle conditions. What do you think?

**Review by jrobinAV — DISMISSED** [review]
NEat! 
And the tests are great too.

**Comment by jrobinAV** [comment]
Well the tests are failing.

File: taipy/core/_orchestrator/_orchestrator_factory.py [hunk 4]
```diff
@@ -87,7 +87,13 @@ def __build_standalone_job_dispatcher(cls, force_restart=False):
     def __build_development_job_dispatcher(cls):
         if isinstance(cls._dispatcher, _StandaloneJobDispatcher):
             cls._dispatcher.stop()
-        cls._dispatcher = _DevelopmentJobDispatcher(typing.cast(_AbstractOrchestrator, cls._orchestrator))
+
+        if util.find_spec(cls._TAIPY_ENTERPRISE_MODULE) is not None:
```
Comment (jrobinAV) [inline taipy/core/_orchestrator/_orchestrator_factory.py]: I guess you ll want to use the _using_enterprise function from common package.

**Review by jrobinAV — DISMISSED** [review]

File: taipy/core/_orchestrator/_orchestrator_factory.py (hunk 4 above)
Comment (toan-quach) [reply taipy/core/_orchestrator/_orchestrator_factory.py]: ah yes, ohh I merged the other PR before this one 🤣

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**